homework/index.js

```diff
@@ -30,15 +15,105 @@
+    }
+    if (repo.description !== null) {
+      appendRow(tBody, 'Description :', repo.description);
+    }
+    appendRow(tBody, 'Forks :', repo.forks);
+    appendRow(tBody, 'Updated :', new Date(repo.updated_at).toLocaleString());
+  }
+
+  async function reposContributor(url) {
+    const root = document.getElementById('root');
+    const rightDiv = document.querySelector('.right-div');
+    createAndAppend('p', rightDiv, {
+      class: 'contributor-header',
+      text: 'Contributions',
     });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const contributors = await response.json();
+
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul, {
+          'aria-label': contributor.login,
+        });
+        const link = createAndAppend('a', li, {
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contributor-item',
+        });
+        createAndAppend('img', link, {
+          src: contributor.avatar_url,
+          height: 48,
+          class: 'contributor-avatar',
+          alt: 'contributor-avatar',
+        });
+        const contributorData = createAndAppend('div', link, { class: 'contributor-data' });
+        createAndAppend('div', contributorData, { text: contributor.login });
+        createAndAppend('div', contributorData, {
+          class: 'contributor-badge',
+          text: contributor.contributions,
+        });
+      });
+    } catch (err) {
+      createAndAppend('p', root, { text: err.message, class: 'alert' });
+    }
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+    const selector = createAndAppend('select', header, { class: 'repo-selector' });
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const repositories = await response.json();
+
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      for (let i = 0; i < repositories.length; i++) {
+        createAndAppend('option', selector, { text: repositories[i].name, value: i });
+      }
+
+      const container = createAndAppend('div', root, { id: 'container' });
+      createAndAppend('div', container, { class: 'left-div' });
+      createAndAppend('div', container, { class: 'right-div' });
+      reposAbstract(selector.value, repositories);
+      reposContributor(repositories[selector.value].contributors_url);
+      selector.addEventListener('change', () => {
+        document.querySelector('.left-div').innerHTML = '';
+        document.querySelector('.right-div').innerHTML = '';
```

Comment:
You can use JavaScript variables rather than looking up the elements in the DOM here:

```js
const container = createAndAppend('div', root, { id: 'container' });
const leftDiv = createAndAppend('div', container, { class: 'left-div' });
const rightDiv = createAndAppend('div', container, { class: 'right-div' });
reposAbstract(selector.value, repositories);
reposContributor(repositories[selector.value].contributors_url);
selector.addEventListener('change', () => {
  leftDiv.innerHTML = '';
  rightDiv.innerHTML = '';
```

Reply:
ok i got it.